### PR TITLE
Added option for downloadview placement.

### PR DESCRIPTION
**Pull request by oed**
I wanted the downloadview at the bottom of the window. So I added an option.

### CODE REVIEW

**Comment by The-Compiler** [comment]
That's the spirit! :)

Two things come to mind:

* I think it'd make more sense to be consistent with `tabs -> position` here instead of using a bool - that'd mean adding a new type (say `HorizontalPosition` or so) to `config/configtypes.py`, which is essentially equal to `Position`, except without a `transform()` and only `north`/`south` as valid values. What do you think?
* All config options are applied immediately in qutebrowser when using `:set` and I'd like to keep it that way - I think the best way would be to unconditionally initialize `DownloadManager`/`DownloadView`/`TabbedBrowser`, and then add a new function which removes them from `self._vbox` (which won't do anything if they don't exist) and then adds them in the right order. Then you can call that in `on_config_changed` if the corresponding section/option was changed.

**Comment by oed** [comment]
1. I thought about that but I didn't think it was a good idea to have another type that just basically a bool. Maybe I'm wrong though.

2. Ah, that makes sense!

File: qutebrowser/config/configdata.py [hunk 1]
```diff
@@ -1006,7 +1010,6 @@ def data(readonly=False):
 
 DATA = data(readonly=True)
 
-
```
Comment (The-Compiler) [inline qutebrowser/config/configdata.py]: nitpick: unrelated whitespace change

File: qutebrowser/mainwindow/mainwindow.py [hunk 2]
```diff
@@ -141,6 +140,24 @@ def on_config_changed(self, section, option):
...
+        if position == 'north':
+            self._vbox.addWidget(self._downloadview)
+            self._vbox.addWidget(self._tabbed_browser)
+        elif position == 'south':
+            self._vbox.addWidget(self._tabbed_browser)
+            self._vbox.addWidget(self._downloadview)
+        else:
+            raise ValueError("Invalid position {}!".format(position))
+        self._vbox.addWidget(self.status)
+
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/mainwindow.py]: another small nitpick: one blank line between methods of a class

File: qutebrowser/mainwindow/mainwindow.py [hunk 3]
```diff
@@ -141,6 +140,24 @@ def on_config_changed(self, section, option):
         """Resize the completion if related config options changed."""
         if section == 'completion' and option in ('height', 'shrink'):
             self.resize_completion()
+        elif section == 'ui' and option == 'downloads-position':
+            self._add_widgets()
+
+    def _add_widgets(self):
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/mainwindow.py]: Can you add a small docstring here?

(I know the docstring checker is rather annoying... but it helps me when trying something and then accidentally forget writing docstrings, so I still think it's better to have a docstring for everything)
